src/js/components/List/List.js

```diff
@@ -349,6 +352,20 @@ const List = React.forwardRef(
                 let content;
                 let boxProps = {};
 
+                const key = getValue(item, index, itemKey) || index;
+
+                let isPinned;
+                if (pinned.length > 0) {
+                  if (typeof item === 'object' && !itemKey) {
+                    console.error(
+                      // eslint-disable-next-line max-len
+                      `Warning: Missing prop itemKey. Prop pin requires itemKey to be specified when data is of type 'object'.`,
+                    );
+                  }
+                  isPinned = pinned?.includes(key);
+                }
+                const pinTextColor = theme.list.item.pinned.text.color;
```

Comment:
It feels a bit odd that this is in `list` theme but it only applies when in DataTableColumns. I thought we were planning on just hard coding the value for now as opposed to having it in theme?

So on line 381 below, just have `inDataColumn ? `text-weak` : undefined`

Reply:
I also wonder if there's anyway to consolidate that DataTableColumns logic in one place. Feels a little odd having it dispersed throughout what otherwise is just generic list code.

Reply:
yeah I can consolidate as well as just have `text-weak` I wasn't sure if we wanted it in theme but agree on naming maybe `dataTableColumns.pinned.text.color`